Document NTP as the canonical clock reference, with yum "HTTPS Error 301" as one symptom (#504)
xcp-ng/xcp-ng-org: docs/management/hosts-pools.md

Correct and consistent clocks across the pool matter more than on ordinary servers: XAPI coordination, live migration, HA heartbeats, logs correlation and Windows guests (which get their initial time from the host) all rely on it. NTP servers are normally configured at [installation time](../installation/install-xcp-ng.md).

> A wrong date rarely fails where you would expect. It usually surfaces as an error that names something else:

Reviewer: ```suggestion
A wrong date rarely fails where you would expect. It usually appears as an error that points to something else:
```

Author: "Rarely fails where you would expect" is the better line, and this is exactly the sentence the page in #514 opens with. I have not applied it here because the paragraph has left this PR: the NTP material was duplicated across both PRs and it now lives only on the dedicated page. Carrying your wording across to #514.

* Repository access stops working. Repositories are served over HTTPS, and a host whose date falls outside the mirror's certificate validity period cannot complete the TLS handshake. `yum` reports the redirect it followed instead of the certificate: see [yum fails with "HTTPS Error 301 - Moved Permanently"](../troubleshooting/common-problems.md#yum-fails-with-https-error-301---moved-permanently).
* The installer refuses the repository keys, with `Signature Key Import Failed`.

> * Hosts that disagree on the time cause problems across the pool, from live migration to HA.

Reviewer: ```suggestion
* Hosts with inconsistent system times cause problems across the pool, from live migration to HA.
```

Author: Agreed, "cause problems across the pool, from live migration to HA" is more concrete than what I had. Same as above, this bullet has moved to `time-synchronization.md` in #514, so I will apply it there rather than reinstate the paragraph here.

On XCP-ng 8.2, the NTP daemon is `ntpd`: edit `/etc/ntp.conf`, then `systemctl restart ntpd` and check with `ntpq -p`. In both cases, `xsconsole` also offers an NTP configuration menu under **Network and Management Interface**.

### Correcting a wrong clock {#correcting-a-wrong-clock}

Check the sources before anything else. `chronyd` can be both running and enabled while having **no time source configured at all**, which is what happens when the date was set manually during installation:

<Terminal shell title="root@xcp-ng-host — Check the clock and its sources">{`
date
systemctl status chronyd
chronyc sources
`}</Terminal>

`210 Number of sources = 0` means the service is working exactly as configured, and doing nothing. Add time sources to `/etc/chrony.conf`:

```
server 0.centos.pool.ntp.org iburst
server 1.centos.pool.ntp.org iburst
server 2.centos.pool.ntp.org iburst
server 3.centos.pool.ntp.org iburst
```

> On an isolated network, use a local time source instead of the public pool. Then restart chronyd and confirm a source is reachable: in `chronyc sources`, a line whose second column is `*` or `+`, not `?`.

Reviewer: ```suggestion
On an isolated network, use a local time source instead of the public pool. Then restart chronyd and verify that a source is reachable: in `chronyc sources`, a line whose marked with `*` or `+` in the second column, rather than `?`.
```

Author: Taking this one, with a note: the marker is on the source line, so "a line whose second column is `*` or `+`" is the part I would keep rather than "a line whose marked with". Applying it on #514 where this paragraph now lives.

> With a reachable source, step the clock and check the result:

Reviewer: ```suggestion
Once a source is reachable, step the clock and check the result:
```

Author: Yes, "step the clock" is the right verb here rather than the longer phrasing I used, especially since the section is about `makestep`. Applying on #514 with the rest.

<Terminal shell title="root@xcp-ng-host — Step the clock">{`
chronyc makestep
date
`}</Terminal>

> `chronyc makestep` is needed because chrony corrects an offset by slewing the clock gradually by default, which never converges for an offset of months or years. It does nothing useful while there are no sources: it reports success and moves the clock nowhere, having no measured offset to step to.

Reviewer: - The first sentence seems a big long, and the two successive adverbs/adverbial phrases ("gradually" and "by default") make it sound a bit heavy. Is there way to split the sentence into two shorter sentences?
- In _"It does nothing useful..."_, does "It" refer to `chronyc makestep`? Since the previous sentence is a bit difficult to read, it's easy to lose track the referenced subject.
- "There is" and "there are" sound very passive and indeterminate. Try to avoid those when possible, when a more concrete subject and verb can be used. I suggest: _"Without any available sources, however, `chronyc makestep` cannot do anything useful: it reports success but does not change the clock because there is no measured offset to apply."_

Author: All three land. The sentence was too long, "It" was ambiguous by the time the reader got to it, and your version is more concrete than mine. Your rewrite is what I will use.

That paragraph now lives on the page in #514, so it gets applied there rather than here. Sorry for making you line-edit the same prose twice: the material existed in both PRs, which is what this rebase fixes.

:::note

> * There is no need to copy the corrected time to the hardware clock by hand. The shipped `/etc/chrony.conf` enables `rtcsync`, so chronyd keeps the hardware clock in step and the correction survives a reboot.

Reviewer: ```suggestion
* There is no need to copy the corrected time to the hardware clock manually. The default `/etc/chrony.conf` enables `rtcsync`, so chronyd keeps the hardware clock synchronized and the corrected time persists across reboots.
```

Author: Taken. "Synchronized" and "persists across reboots" read better than my "in step" and "survives a reboot", and this note has moved to the page in #514 along with the rest of the chrony material.
